//MessageType
syntax = "proto3";
package Protobuf;
option csharp_namespace = "Communication.Proto";

enum MessageType
{
    AddPlayer = 0;
    Move = 1;
    Attack = 2;         //注：所有人物都用子弹攻击
    Pick = 3;
    UseProp = 4;
    ThrowProp = 5;
    Send = 6;
    InvalidPlayer = 7;
    ValidPlayer = 8;
    StartGame = 9;
    Gaming = 10;
    EndGame = 11;
    UseCommonSkill = 12;   //使用CommonSkill
    UseUltimateSkill = 13;   //使用UltimateSkill。UltimateSkill极有可能不会有，即人物极有可能只有PassiveSkill和CommonSkill。
    UseGem=14;
    ThrowGem=15;
    Map=16;
}

enum GameObjType
{
    NullGameObjType = 0;
    Character = 1;
    Wall = 2;
    Prop = 3;
    Bullet = 4;
    BirthPoint = 5;
    OutOfBoundBlock = 6;
    Grass = 7;
    GemWell=8;
    BombedBullet=9;
}
enum ShapeType
{
    NullShapeType = 0;
    Circle = 1;     //仍然，子弹和人物为圆形，格子为方形
    Square = 2;
}
enum PlaceType  //位置标志，包括陆地，草丛，以及角色技能带来的隐身。游戏中每一帧都要刷新各个物体的该属性
{
    NullPlaceType = 0;
    Land = 1;
    Grass1 = 2;
    Grass2 = 3;
    Grass3 = 4;
}
enum PropType    // 道具的类型
{
    NullPropType = 0;
    addSpeed = 1; // 加速
    addLIFE = 2; // 多一条命
    Gem = 3;    // 新增：宝石
    Shield = 4; // 护盾
    Spear = 5; // 破盾
}
enum PassiveSkillType   // 被动技能，有名字的都是已经实现了的
{
    NullPassiveSkillType = 0;
    RecoverAfterBattle = 1;
    SpeedUpWhenLeavingGrass = 2;
    Vampire = 3;
    PSkill3 = 4;
    PSkill4 = 5;
    PSkill5 = 6;
}
enum ActiveSkillType    // 主动技能，有名字的都是已经实现了的
{
    NullActiveSkillType = 0;
    BecomeVampire = 1;
    BecomeAssassin = 2;
    NuclearWeapon = 3;
    SuperFast = 4;
    ASkill4 = 5;
    ASkill5 = 6;
}
enum BuffType    //buff
{
    NullBuffType = 0;
    MoveSpeed = 1;
    AddLIFE = 2;
    ShieldBuff = 3;
    SpearBuff = 4;
}
enum BulletType  //会有不同类型的子弹，现在只实现了“原子弹”
{
    NullBulletType = 0;
    LineBullet=1;
    CommonBullet2=2;
    FastBullet=3;
    OrdinaryBullet=4;
    AtomBomb=5;
}
//先偷懒把枚举类型粘在这里